### PR TITLE
chore(deps): bump phpoffice/phpspreadsheet 5.4.0 → 5.7.0

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

Resolves five open Dependabot alerts on `composer.lock` for `phpoffice/phpspreadsheet`:

| Alert | Severity | CVE | Issue |
|---|---|---|---|
| [#96](https://github.com/netresearch/timetracker/security/dependabot/96) | High | CVE-2026-34084 | SSRF/RCE in `IOFactory::load` when `$filename` is user-controlled |
| [#97](https://github.com/netresearch/timetracker/security/dependabot/97) | High | CVE-2026-40863 | CPU DoS via unbounded row index in SpreadsheetML XML reader |
| [#98](https://github.com/netresearch/timetracker/security/dependabot/98) | High | CVE-2026-40902 | CPU DoS via unbounded row number in XLSX row dimensions |
| [#94](https://github.com/netresearch/timetracker/security/dependabot/94) | Medium | CVE-2026-35453 | XSS via `NumberFormat` `@` text substitution (HTML writer) |
| [#95](https://github.com/netresearch/timetracker/security/dependabot/95) | Medium | CVE-2026-40296 | XSS via `@` text placeholder bypass of `htmlspecialchars` (HTML writer) |

Bump stays within existing `^5.0` constraint in `composer.json`.

### Application impact assessment

- **CVE-2026-34084 (SSRF/RCE):** not exploitable on `main` — `src/Controller/Controlling/ExportAction.php:115` uses `IOFactory::createReader('Xlsx')` with a hardcoded reader type, never `IOFactory::load($userPath)`. Bumping removes the foot-gun for future code.
- **DoS / XSS CVEs:** the export pipeline produces XLSX via PhpSpreadsheet, so the writer-side issues were applicable.

### Test fixture update

`Coordinate::columnIndexFromString()` now hard-rejects column references beyond `XFD` (16384, the Excel format limit). The previous `LOReadFilterTest` data provider included `ZZZ` (resolves to 18278), which is unreachable in real spreadsheets — PhpSpreadsheet itself never iterates past `XFD`. Replaced with `XFD` as the boundary case. Production `LOReadFilter` is unchanged.

## Test plan

- [x] `composer update phpoffice/phpspreadsheet --with-dependencies` → bumped to 5.7.0
- [x] Export-related PHPUnit suite (Service / Repository / Dto / Util / Controller): **85 tests, 163 assertions, 0 failures**
- [ ] Full CI suite (handled by GHA on this PR)
- [ ] Manual smoke: open an exported XLSX in LibreOffice (column-limit workaround behavior unchanged)

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -12629,5 +12629,5 @@
     "platform-overrides": {
         "php": "8.5"
     },
-    "plugin-api-version": "2.9.0"
+    "plugin-api-version": "2.6.0"
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply composer.lock:12632]: The downgrade is intentional and aligns the lock with the project's pinned Composer.

`docker-bake.hcl:27` pins `COMPOSER_IMAGE = "composer:2.8"`, and the dev/tools/prod images in `Dockerfile` all copy that binary. `composer:2.8` (currently 2.8.12) advertises `plugin-api-version: 2.6.0`. Verified inside the dev image:

```
$ docker compose run --rm app-dev composer --version
Composer version 2.8.12 2025-09-19 13:41:59
```

The previous `2.9.0` in the lock was produced by a Composer 2.9.x build run outside this image, which was inconsistent with the project's pin. Re-running `composer update` in the canonical Docker dev image (commit 3d099cc8) restores the value that matches `composer:2.8`. `plugin-api-version` is informational, but keeping it consistent with the pinned binary is the right thing here. Bumping the project to Composer 2.9 is out of scope for a security patch.
